### PR TITLE
docker-buildx: symlink plugin to brew owned cli-plugins

### PR DESCRIPTION
**Pull request by narqo**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

The PR updates the existing formula with `docker-buildx` Docker plugin, adding a post install step, that links the plugin into `#{HOMEBREW_PREFIX}/lib/docker/cli-plugins` directory. Users are expected to add the directory to the list of [`cliPluginsExtraDirs`][1] in their docker-cli's config.

E.g. this is how it looks in my config.json

```
% cat ~/.config/docker/config.json
{
    "auths": {},
    "currentContext": "colima",
    "cliPluginsExtraDirs": [
        "/opt/homebrew/lib/docker/cli-plugins"
    ]
}
```

I find this behaviour much more in-line with how managed packages should work. That is, as a user I find it more natural to update my local config, to point to a managed directory with plugins, than symlinking the managed plugin into my home directory (i.e. the previous suggestion from "caveats").

---

Note on Docker's `cliPluginsExtraDirs`, this parameter was added to docker-cli more than 4 years ago, but the fields of the config aren't very well documented it seems.

[1]: https://pkg.go.dev/github.com/docker/cli@v25.0.3+incompatible/cli/config/configfile#ConfigFile.CLIPluginsExtraDirs

### CODE REVIEW

File: Formula/d/docker-buildx.rb [hunk 1]
```diff
@@ -32,11 +33,18 @@ def install
     generate_completions_from_executable(bin/"docker-buildx", "completion")
   end
 
+  def post_install
+    cli_plugins = HOMEBREW_PREFIX/"lib/docker/cli-plugins"
```
Comment (carlocab) [inline Formula/d/docker-buildx.rb:37]: Could you motivate this choice of path? Is `lib/docker/cli-plugins` a "well-known" path (in some sense)?

File: Formula/d/docker-buildx.rb [hunk 2]
```diff
@@ -32,11 +33,18 @@ def install
     generate_completions_from_executable(bin/"docker-buildx", "completion")
   end
 
+  def post_install
```
Comment (carlocab) [inline Formula/d/docker-buildx.rb:36]: We could probably replace this `post_install` entirely by doing something like
```ruby
(lib/"docker/cli-plugins").install_symlink bin/"docker-buildx"
```
Any reason why we're not doing that?

File: Formula/d/docker-buildx.rb (hunk 1 above)
Comment (narqo) [reply Formula/d/docker-buildx.rb:37]: Yes, following the original design https://github.com/docker/cli/issues/1534 docker-cli scans these paths, when it searches for plugins:

> Unix-like OSes:
> - `$HOME/.docker/cli-plugins`
> - `/usr/local/lib/docker/cli-plugins` & `/usr/local/libexec/docker/cli-plugins`
> - `/usr/lib/docker/cli-plugins` & `/usr/libexec/docker/cli-plugins`

I think it's reasonable to keep Docker CLI's plugins, that brew manages, in the similar structure.

Note, Docker Desktop seems to amend the default list of the paths and bundles several cli-plugins into the distribution itself (e.g. buildx and compose live inside the docker desktop's bundle).

File: Formula/d/docker-buildx.rb (hunk 2 above)
Comment (narqo) [reply Formula/d/docker-buildx.rb]: That's indeed simpler; I updated the changes. PHAL

**Review by chenrui333 — APPROVED** [review]

**Comment by chenrui333** [comment]
@narqo thanks for your first contribution to homebrew-core! 🎉

**Pull request merged**